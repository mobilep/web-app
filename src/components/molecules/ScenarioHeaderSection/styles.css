.ScenarioHeaderSection {
	display: flex;
	padding: var(--scenario-section-vertical-padding) 0;
}

.ScenarioHeaderSection-content {
	flex-grow: 1;
	position: relative;
	width: calc(100% - 431px);
	display: flex;
	flex-direction: column;
}

.ScenarioHeaderSection-header {
	font-size: 22px;
	color: var(--almostBlack);
	margin-bottom: 20px;
}

.ScenarioHeaderSection-text {
	font-size: 14px;
	margin-bottom: 30px;
	color: var(--almostBlack);
}

.ScenarioHeaderSection-image .ScenarioHeaderSection-video-tile {
	background: transparent;
}

.ScenarioHeaderSection-image .ScenarioHeaderSection-video-tile:disabled {
	background-color: var(--grey-blur);
}

.ScenarioHeaderSection input.ScenarioHeaderSection-input-name {
	height: 30px;
	font-weight: 600;
	font-size: 22px;
}

.ScenarioHeaderSection input.ScenarioHeaderSection-input-name::placeholder {
	font-size: 22px;
	font-weight: 600;
}

div.ScenarioHeaderSection-inputWrapper {
	width: 80%;
}

@media (min-width: 769px) {
	.ScenarioHeaderSection-mobile {
		display: none;
	}
}

@media (max-width: 768px) {
	.ScenarioHeaderSection {
		display: none;
	}
}

@media (max-width: 1200px) {
	.ScenarioHeaderSection {
		flex-wrap: wrap-reverse;
	}

	div.ScenarioHeaderSection-inputWrapper {
		width: 91%;
	}
}
