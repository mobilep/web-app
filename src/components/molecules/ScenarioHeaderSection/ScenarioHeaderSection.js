import React, { Component, Fragment } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { ScenarioUserCounter, VideoTile } from '../../atoms';
import { ScenarioTextInput, ScenarioMobileHeaderSection } from '..';
import { TILE_TYPES } from '../../atoms/VideoTile/constants';
import constants from '../../../constants';
import './styles.css';
const { common } = constants;
const { mediaLoadingStates } = common;
const { modes: { EDIT_MODE, CREATE_MODE, PRACTICE_MODE } } = common;

class ScenarioHeaderSection extends Component {

	static propTypes = {
		addText: PropTypes.string,
		addVideoText: PropTypes.string,
		authorDetails: PropTypes.object,
		buttonPrimaryText: PropTypes.string,
		buttonSecondaryText: PropTypes.string,
		canEditVideo: PropTypes.bool,
		changeVideoText: PropTypes.string,
		count: PropTypes.number,
		counterText: PropTypes.string,
		createMode: PropTypes.bool,
		description: PropTypes.string,
		descriptionPlaceholder: PropTypes.string,
		draft: PropTypes.bool,
		fallBackVideoUrl: PropTypes.string,
		header: PropTypes.string,
		match: PropTypes.object,
		mode: PropTypes.string,
		namePlaceholder: PropTypes.string,
		onDescriptionChange: PropTypes.func,
		onDraftClick: PropTypes.func,
		onHeaderChange: PropTypes.func,
		onPrimaryClick: PropTypes.func,
		onSecondaryClick: PropTypes.func,
		onVideoChange: PropTypes.func,
		practiceMode: PropTypes.bool,
		saveDraftText: PropTypes.string,
		scenarioType: PropTypes.string,
		showValidationDescriptionError: PropTypes.bool,
		showValidationNameError: PropTypes.bool,
		thumbnailUrl: PropTypes.string,
		updateDate: PropTypes.string,
		validationDescriptionError: PropTypes.string,
		validationNameError: PropTypes.string,
		videoId: PropTypes.string,
		videoState: PropTypes.string,
		videoUrl: PropTypes.string,
	}

	state = {
		VIDEO_TILE_STATUS: TILE_TYPES.ADD,
	}

	handleHeaderChange = ({ target }) => {
		const { value } = target;
		this.props.onHeaderChange(value);
	}

	handleDescriptionChange = ({ target }) => {
		const { value } = target;
		this.props.onDescriptionChange(value);
	}

	isContentEditable ({ mode } = this.props) {
		return mode === EDIT_MODE || mode === CREATE_MODE;
	}

	handleVideoChange = (fileId) => {
		this.props.onVideoChange(fileId);
	}

	getVideoTileType () {
		const { videoState, videoId } = this.props;
		if (videoId && videoState !== mediaLoadingStates.COMPLETED) {
			return TILE_TYPES.UPLOADING;
		} else if (this.isContentEditable()) {
			return TILE_TYPES.ADD;
		} else {
			return TILE_TYPES.DEFAULT;
		}
	}

	getVideoTileButtonClassName () {
		const { mode, thumbnailUrl } = this.props;
		if ((mode === CREATE_MODE && thumbnailUrl) || mode === EDIT_MODE) return 'ScenarioHeaderSection-video-tile';
		return '';
	}

	render () {
		const {
			mode,
			count,
			counterText,
			description,
			descriptionPlaceholder,
			header,
			namePlaceholder,
			onDraftClick,
			thumbnailUrl,
			videoUrl,
			fallBackVideoUrl,
			draft,
			saveDraftText,
			videoId,
			showValidationNameError,
			showValidationDescriptionError,
			validationNameError,
			validationDescriptionError,
			updateDate,
			authorDetails,
			addVideoText,
			changeVideoText,
			canEditVideo,
		} = this.props;

		let addText;

		if (this.isContentEditable()) {
			if (!videoId) {
				addText = addVideoText;
			} else {
				addText = changeVideoText;
			}
		}
		const plusIcon = mode === CREATE_MODE && !draft;
		const videoTileProps = {
			authorDetails,
			videoId,
			videoUrl,
			fallBackVideoUrl,
			onVideoSend: this.handleVideoChange,
			thumbnailUrl,
			buttonClassName: this.getVideoTileButtonClassName(),
			type: this.getVideoTileType(),
			addText,
			plusIcon,
			canEditVideo,
		};

		if (!videoId && mode !== CREATE_MODE) {
			videoTileProps.withAuthorAvatar = true;
		}

		return (
			<Fragment>
				<ScenarioMobileHeaderSection
					{...this.props}
					canEditVideo={canEditVideo}
					videoTileProps={videoTileProps}
					className='ScenarioHeaderSection-mobile'
					contentEditable={this.isContentEditable()}
					onDescriptionChange={this.handleDescriptionChange}
					onHeaderChange={this.handleHeaderChange}
					onVideoSend={this.handleVideoChange}
					saveDraftText={saveDraftText}
					onDraftClick={onDraftClick}
					updateDate={updateDate}
				/>
				<div className='ScenarioHeaderSection'>
					<div className='ScenarioHeaderSection-content'>
						<div className='ScenarioHeaderSection-header'>
							{this.isContentEditable()
								? <ScenarioTextInput
									showError={showValidationNameError}
									errorText={validationNameError}
									onChange={this.handleHeaderChange}
									value={header}
									placeholder={namePlaceholder}
									name='name'
									className='ScenarioHeaderSection-inputWrapper'
									maxLength={128}
									inputClassName='ScenarioHeaderSection-input-name' />
								: header}
						</div>
						<div className='ScenarioHeaderSection-text'>
							{this.isContentEditable()
								? <ScenarioTextInput
									showError={showValidationDescriptionError}
									errorText={validationDescriptionError}
									onChange={this.handleDescriptionChange}
									value={description}
									placeholder={descriptionPlaceholder}
									className='ScenarioHeaderSection-inputWrapper'
									maxLength={128}
									name='description' />
								: description}
						</div>
						{mode !== PRACTICE_MODE && count
							? <ScenarioUserCounter
								className='ScenarioHeaderSection-counter'
								count={count}
								text={counterText} />
							: null}
					</div>
					<div className='ScenarioHeaderSection-image'>
						<VideoTile {...videoTileProps} />
					</div>
				</div>
			</Fragment>
		);
	}
}

export default withRouter(ScenarioHeaderSection);
