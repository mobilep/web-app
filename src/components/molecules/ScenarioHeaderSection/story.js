import React from 'react';
import { storiesOf } from '@storybook/react';
import { selectV2 } from '@storybook/addon-knobs';
import constants from '../../../constants';
import ScenarioHeaderSection from './ScenarioHeaderSection';

const { common } = constants;
const { modes: { EDIT_MODE, CREATE_MODE, PRACTICE_MODE } } = common;

storiesOf('2. Molecules / ScenarioHeaderSection', module)
	.add('default', () => (<ScenarioHeaderSection
		thumbnailUrl='http://via.placeholder.com/350x150/00423b'
		header='Discovery Questioning Skills'
		description='Practice asking discovery questions when meeting someone for the first time.'
		buttonSecondaryText='Delete'
		buttonPrimaryText='Edit'
		mode={selectV2('modes', {
			edit: EDIT_MODE,
			create: CREATE_MODE,
			practice: PRACTICE_MODE,
		}, CREATE_MODE)}
		count={5}
		counterText='people'
		addText='Add video'
		namePlaceholder='Give the scenario a name…'
		descriptionPlaceholder='Describe the scenario…'
	/>));
