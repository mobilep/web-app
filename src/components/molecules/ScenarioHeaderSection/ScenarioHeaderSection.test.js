import React from 'react';
import { shallow } from 'enzyme';

import ScenarioHeaderSection from './ScenarioHeaderSection';

describe('Molecule - ScenarioHeaderSection', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<ScenarioHeaderSection
				thumbnailUrl='URL'
				header='Header'
				description='Description'
				buttonCancelText='Cancel'
				buttonPrimaryText='Primary'
				count={5}
				counterText='Counter'
				practiceMode
				editMode
				createMode
				addText='Add'
				namePlaceholder='Name'
				descriptionPlaceholder='Describe'
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
