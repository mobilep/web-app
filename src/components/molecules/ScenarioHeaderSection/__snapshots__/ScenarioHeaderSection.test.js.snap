// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - ScenarioHeaderSection Should render 1`] = `
<Route>
  <Component />
</Route>
`;
